### PR TITLE
checkpoint-1.8.2

### PR DESCRIPTION
**Pull request by arvidn**
this merges 1.8.2 into main. There were conflicts in these files:

```
Auto-merging chia/full_node/full_node.py
Auto-merging chia/timelord/timelord_api.py
Auto-merging setup.py
CONFLICT (content): Merge conflict in setup.py
Auto-merging tests/core/full_node/stores/test_full_node_store.py
CONFLICT (content): Merge conflict in tests/core/full_node/stores/test_full_node_store.py
Auto-merging tests/core/full_node/test_full_node.py
Automatic merge failed; fix conflicts and then commit the result.
```

Which were manually patched up. Reviewers should probably focus on those.

### CODE REVIEW

**Review by altendky — CHANGES_REQUESTED** [review]
just putting a hold on this until it seems clear we understand what's going on

**Comment by arvidn** [comment]
it looks like Amine will fix up the original checkpoint merge instead

File: tests/core/full_node/stores/test_full_node_store.py [hunk 1]
```diff
@@ -265,7 +304,9 @@ async def test_basic_store(
                 result = await blockchain.get_sp_and_ip_sub_slots(block.header_hash)
                 assert result is not None
                 sp_sub_slot, ip_sub_slot = result
```
Comment (almogdepaz) [inline tests/core/full_node/stores/test_full_node_store.py:306]: missing 

  next_sub_slot_iters, next_difficulty = get_next_sub_slot_iters_and_difficulty(
                    blockchain.constants, False, sb, blockchain
                )
in line 306

Comment (altendky) [reply tests/core/full_node/stores/test_full_node_store.py:306]: Thanks for looking this over.  I believe @AmineKhaldi noted this before as well and Arvid.  I'm glad you all are in agreement about it.  `:]`  We have now merged Amine's PR for this and I think this one can be closed.

**Pull request closed without merging**